client/js/app.js

```diff
@@ -0,0 +1,56 @@
+const btn = document.querySelector('#post-form');
+const main = document.querySelector('main');
+
+function submitPost(e){
+    e.preventDefault();
+    const postData = {
+        title: e.target.title.value,
+        name: e.target.name.value,
+        body: e.target.body.value
+    }
+
+    const options = {
+        method: 'POST',
+        body: JSON.stringify(postData),
+        headers: {
+            "Content-Type": "application/json"
+        }
+    };
+
+    fetch('http://localhost:3000/posts/', options)
+        .then(d=>d.json())
+        .then(d => window.location.hash = `/posts/${d.id}`)
```

Comment:
I think you didn't have to put the `/` before `posts` just because it ends up looking like `/#/posts`, but that's just a personal preference